influxql/engine.go

```diff
@@ -138,7 +134,18 @@ func (m *MapReduceJob) Execute(out chan *Row, filterEmptyResults bool) {
 	startTimeBucket := m.TMin / m.interval * m.interval
 
 	for i, _ := range resultTimes {
-		t := startTimeBucket + (int64(i) * m.interval * int64(m.stmt.Offset+1))
+		var t int64
+		if m.stmt.Offset > 0 {
+			t = startTimeBucket + (int64(i+1) * m.interval * int64(m.stmt.Offset))
+		} else {
+			t = startTimeBucket + (int64(i+1) * m.interval) - m.interval
+		}
+
+		// If we start getting out of our max time range, then truncate values and return
+		if t > m.TMax && !isRaw {
```

Comment:
Would this ever happen? Shouldn't the size of `resultTimes` be fixed to the exact number of buckets we want?